Autostart update to volumio 3 thomas doerr

thomasdoerr opened the pull request:
Hi,

I migrated the AutoStart Plugin which existed in volumio 2. I just noticed today, that it wasn't ported to the new version.

I tested it on my 2 devices.

Thanks,
Thomas

balbuze commented:
hello,
please remove from install.sh
sudo apt-get update
# Install the required packages via apt-get
sudo apt-get -y install
as it does nothing.
Re submit please
Thanks

balbuze commented:
please, update version number and
- send a new github PR
- submit the plugin with 'volumio plugin submit'
The submitted plugin is not in phase with PR!

thomasdoerr commented:
Hello @balbuze,

I just double checked on my development raspery pi. Changed one setting after the another, rebooted in the meantime, both changes are reflected.
After rebooting, both of my radios start playing. One of the radios is used every morning and works as expected.

Perhaps, the submitted plugin did not reflect the latest code. I thought I submitted it but...perhaps not

I changed the version and submitted it again. Would you please test it again and send me more details if it doesn't work?

Have a nice rest of the weekend
Thomas

balbuze commented:
Hello Thomas,
please have look here : https://developers.volumio.com/plugins/index-js
OnVolumioStart is not conform.
And saved settings are not kept in UI. Change value, save, refresh the page. It is reset to default.
TY

thomasdoerr commented:
Good Morning @balbuze, 

sorry for the delay. I wasn't able to spend time on that. I'm adjusting the code. Just understood what you mean. There is too much code in OnVolumioStart.
Let me rework on that and come back to you, once I've done it. Not sure if this will be today :-)
Best,
Thomas

thomasdoerr commented:
done...I'll in parallel continue testing on my machines.
Best,
Thomas

balbuze commented:
Helllo!
Looks better! Please submit the plugin, I'll have look.
Thanks

thomasdoerr commented:
I submitted it just before pushing the code. Do you see it? Otherwise I’ll check it again tomorrow.
Have a nice evening! Thomas

On 29. Oct 2022, at 18:12, balbuze ***@***.***> wrote:

﻿

Helllo!
Looks better! Please submit the plugin, I'll have look.
Thanks

—
Reply to this email directly, view it on GitHub<https://github.com/volumio/volumio-plugins-sources/pull/146#issuecomment-1295886476>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABUZMBKZ3XPZ2T2TW4UOMHLWFVEFBANCNFSM6AAAAAARAZ6G44>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

thomasdoerr commented:
Good Morning @balbuze ,

I just double checked it and found this in my history:
volumio@volumio:~/volumio-plugins-sources/autostart$ volumio plugin submit
....
? do you want to change your version? (leave blank for default) 3.0.3

Another thing: you mentioned, that the settings dialog behaves not correctly. I checked that yesterday and noticed that the settings are saved by volumio into data/configuration. As far as I tested it, the oposite also works: settings from the json file are displayed in the UI. Please let me know, if you find any issues in that area. 
Also pushed a tiny fix for the documentaion: https://github.com/volumio/volumio-developers-docs/pull/4
Have a nice rest of the weekend,
Thomas

balbuze commented:
please submit v3.0.3 related to your last PR because I see only v3.0.2
TY

thomasdoerr commented:
volumio@volumio:~/volumio-plugins-sources/autostart$ volumio plugin submit

This command will submit a plugin for publishing in the plugins store. Newly submitted plugins will be verified by the volumio team. After verification the plugin will be publically available on the beta channel. When the plugin is properly tested in the beta channel, it will be promoted to the stable channel, available to download by everyone. Every subsequent version of the plugin will alo be put in the beta channel, and will be promoted to the stable channel after testing.

? do you want to change your version? (leave blank for default) 3.0.3
Compressing the plugin
No modules found, running "npm install"
Plugin succesfully compressed
Plugin added
Plugin uploaded

-> Done

thomasdoerr commented:
Do you want package-lock.json to be committed? It is contained in the .gitignore in the plugin folder but not in that on the root folder. That might be the reason why it has been committed here.
Regards,
Thomas

thomasdoerr commented:
Hello @balbuze ,

is there anything you need from my side to merge this PR? Just noticed, that it is not merged and I'm not aware of anything I need to provide. 

Have a nice weekend!
Thomas

thomasdoerr commented:
Hi @balbuze, 

version 3.0.4 fixes the active / inactive problem, mentioned in the forum. 

Best,
Thomas

pmcculey commented:
Hope I'm commenting in the right place!
I installed the autostart plugin as I wanted to resume play of audiobooks at the position where I left off. However play resumes at the start of the book. Is there any way to change that? Or is it only possible to resume at the start of the last track played?

Thanks,

Joe

thomasdoerr commented:
Hi Joe, no I think, there is no way to change that. It has never been tested with playbooks. The plugin has been implemented to play again the last Radio Channel when you start the pi.

Best,
Thomas

On 10. Mar 2025, at 22:00, pmcculey ***@***.***> wrote:


Hope I'm commenting in the right place!
I installed the autostart plugin as I wanted to resume play of audiobooks at the position where I left off. However play resumes at the start of the book. Is there any way to change that? Or is it only possible to resume at the start of the last track played?

Thanks,

Joe

—
Reply to this email directly, view it on GitHub<https://github.com/volumio/volumio-plugins-sources/pull/146#issuecomment-2711811934>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABUZMBNDHWSYB6NSODQSRSL2TX4NFAVCNFSM6AAAAABYXHOD2SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMJRHAYTCOJTGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

[pmcculey]pmcculey left a comment (volumio/volumio-plugins-sources#146)<https://github.com/volumio/volumio-plugins-sources/pull/146#issuecomment-2711811934>

Hope I'm commenting in the right place!
I installed the autostart plugin as I wanted to resume play of audiobooks at the position where I left off. However play resumes at the start of the book. Is there any way to change that? Or is it only possible to resume at the start of the last track played?

Thanks,

Joe

—
Reply to this email directly, view it on GitHub<https://github.com/volumio/volumio-plugins-sources/pull/146#issuecomment-2711811934>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABUZMBNDHWSYB6NSODQSRSL2TX4NFAVCNFSM6AAAAABYXHOD2SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMJRHAYTCOJTGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

pmcculey commented:
Welll thanks for the reply anyhow Thomas.

Best,

Joe
